sdks/python/apache_beam/runners/portability/fn_api_runner/translations.py

```diff
@@ -557,8 +557,13 @@ def add_parent(child, parent):
             pipeline_proto.components.transforms[parent])
         copy_output_pcollections(components.transforms[parent])
         del components.transforms[parent].subtransforms[:]
-        add_parent(parent, parents.get(parent))
+      # Ensure that child is the lsat item in the parent's subtransforms.
+      # This is required to maintain topological order with sort_stages.
```

Comment:
Isn't sort_stages called before this? Or did you mean this is required to maintain the topolicial order from sort_stages?

Reply:
Rephrased to make this clearer (topological order is maintained only if sort_stages was previously called).